File: Arrays/Solutions/Maximize_Area_of_Square_Hole.cpp
class Solution {
public:
    // Helper function to find longest consecutive streak
    int longestConsecutiveStreak(vector<int>& bars) {
        sort(bars.begin(), bars.end());

        int maxStreak = 1;
        int currStreak = 1;

        for (int i = 1; i < bars.size(); i++) {
            // Check if bars are consecutive
            if (bars[i] == bars[i - 1] + 1) {
                currStreak++;
            } else {
                currStreak = 1;
            }
            maxStreak = max(maxStreak, currStreak);
        }

        return maxStreak;
    }

    int maximizeSquareHoleArea(int n, int m, vector<int>& hBars, vector<int>& vBars) {
        // Longest consecutive removable bars
        int maxH = longestConsecutiveStreak(hBars) + 1;
        int maxV = longestConsecutiveStreak(vBars) + 1;

        // Largest possible square side
        int side = min(maxH, maxV);

        return side * side;
    }
};

File: Arrays/Solutions/Maximize_Area_of_Square_Hole.py
class Solution:
    def maximizeSquareHoleArea(
        self,
        n: int,
        m: int,
        hBars: list[int],
        vBars: list[int]
    ) -> int:

        # Helper to find longest consecutive streak
        def longest_consecutive_streak(bars: list[int]) -> int:
            bars.sort()

            max_streak = 1
            curr_streak = 1

            for i in range(1, len(bars)):
                # If bars are consecutive, extend streak
                if bars[i] == bars[i - 1] + 1:
                    curr_streak += 1
                else:
                    curr_streak = 1

                max_streak = max(max_streak, curr_streak)

            return max_streak

        # Compute max gaps in both directions
        max_height = longest_consecutive_streak(hBars) + 1
        max_width = longest_consecutive_streak(vBars) + 1

        # Largest square side
        side = min(max_height, max_width)

        return side * side
